src/main/java/com/labsynch/labseer/chemclasses/bbchem/ChemStructureServiceBBChemImpl.java

```diff
@@ -85,48 +77,71 @@ public JdbcTemplate getJdbcTemplate() {
+					query = BBChemDryRunStructure.findBBChemDryRunStructuresByPreRegEquals(serviceBBChemStructure.getPreReg());
+				} else {
+					throw new CmpdRegMolFormatException("Structure type not implemented for BBChem searches on " + structureType);
+				}
+			} else if (searchType == SearchType.DUPLICATE_TAUTOMER | searchType == SearchType.EXACT){
+				// DUPLICATE TAUTOMER hashes are stored on the reg column
+				if(structureType == StructureType.PARENT) {
+					query = BBChemParentStructure.findBBChemParentStructuresByRegEquals(serviceBBChemStructure.getReg());
+				} else if (structureType == StructureType.SALT) {
+					query = BBChemSaltStructure.findBBChemSaltStructuresByRegEquals(serviceBBChemStructure.getReg());
+				} else if (structureType == StructureType.SALT_FORM) {
+					query = BBChemSaltFormStructure.findBBChemSaltFormStructuresByRegEquals(serviceBBChemStructure.getReg());
+				} else if (structureType == StructureType.DRY_RUN) {
+					query = BBChemDryRunStructure.findBBChemDryRunStructuresByRegEquals(serviceBBChemStructure.getReg());
+				} else {
+					throw new CmpdRegMolFormatException("Structure type not implemented for BBChem searches on " + structureType);
+				}
 			} else {
-				throw new CmpdRegMolFormatException("Structure type not implemented for BBChem searches" + structureType);
+				throw new CmpdRegMolFormatException("Structure type not implemented for BBChem searches on " + structureType);
 			}
-		}else if (searchType == SearchType.DUPLICATE_TAUTOMER | searchType == SearchType.EXACT){
-			// DUPLICATE TAUTOMER hashes are stored on the reg column
-			if(structureType == StructureType.PARENT) {
-				query = BBChemParentStructure.findBBChemParentStructuresByRegEquals(serviceBBChemStructure.getReg());
-			} else if (structureType == StructureType.SALT) {
-				query = BBChemSaltStructure.findBBChemSaltStructuresByRegEquals(serviceBBChemStructure.getReg());
-			} else if (structureType == StructureType.SALT_FORM) {
-				query = BBChemSaltFormStructure.findBBChemSaltFormStructuresByRegEquals(serviceBBChemStructure.getReg());
-			} else if (structureType == StructureType.DRY_RUN) {
-				query = BBChemDryRunStructure.findBBChemDryRunStructuresByRegEquals(serviceBBChemStructure.getReg());
+
+			// int is not nullable so you either have a valid maxResults passed in
+			// or -1 to express unlimited
+			if(maxResults > -1) {
+				query.setMaxResults(maxResults);
+			}
+
+			bbChemStructures = query.getResultList();
+			
+		} else if (searchType == SearchType.SUBSTRUCTURE | searchType == SearchType.SIMILARITY){
+			// Calculate fingerprint
+			BBChemParentStructure serviceBBChemStructure = bbChemStructureService.getProcessedStructure(molfile, true);
+			if(searchType == SearchType.SUBSTRUCTURE) {
+				if(structureType == StructureType.PARENT) {
+					bbChemStructures = BBChemParentStructure.findBBChemParentStructuresBySubstructure(serviceBBChemStructure.getSubstructure(), maxResults);
```

Comment:
I know you know this already, but just leaving a reminder that we should be following up with another call to BBChem to do a true subgraph substructure search here.

Reply:
Updated to call the service. Tested and it works for me.